components/clubhouse_io/actions/create-story/create-story.js

```diff
@@ -0,0 +1,371 @@
+      type: "boolean",
+      label: "Archived",
+      description: "Controls the story’s archived state.",
+      optional: true,
+    },
+    comments: {
+      type: "string",
+      label: "Comments",
+      description:
+        "An array with comment objects to add to the story. Each comment object must have the [CreateStoryCommentParams](https://clubhouse.io/api/rest/v3/#CreateStoryCommentParams) structure. Alternatively, provide a string that will `JSON.parse` to an array of comment objects. ",
+      optional: true,
+    },
+    completedAtOverride: {
+      type: "string",
+      label: "Completed at Override Date",
+      description:
+        "A manual override for the time/date the Story was completed.",
+      optional: true,
+    },
+    createdAt: {
+      type: "string",
+      label: "Created at Date",
+      description: "The time/date the Story was created.",
+      optional: true,
+    },
+    dueDate: {
+      type: "string",
+      label: "Due Date",
+      description: "The due date of the story.",
+      optional: true,
+    },
+    description: {
+      type: "string",
+      label: "Description",
+      description: "The description of the story.",
+      optional: true,
+    },
+    epicId: {
+      type: "integer",
+      label: "Epic ID",
+      description: "The unique identifier of the epic the story belongs to.",
+      async options() {
+        const options = [];
+        const epics = await this.clubhouse.listEpics();
+        epics.forEach((epic) => {
+          options.push({
+            label: epic.name,
+            value: epic.id,
+          });
+        });
+        return options;
+      },
+      optional: true,
+    },
+    estimate: {
+      type: "integer",
+      label: "Estimate",
+      description:
+        "The numeric point estimate of the story. Can also be null, which means unestimated.",
+      optional: true,
+    },
+    externalId: {
+      type: "integer",
+      label: "External Id",
+      description:
+        "This field can be set to another unique ID. In the case that the Story has been imported from another tool, the ID in the other tool can be indicated here.",
+      optional: true,
+    },
+    externalLinks: {
+      type: "string[]",
+      label: "External Links",
+      description: "A string array of External Links associated with this story.",
+      optional: true,
+    },
+    fileIds: {
+      type: "integer[]",
+      label: "File Ids",
+      description: "An integer array of IDs of files attached to the story.",
+      optional: true,
+    },
+    followerIds: {
+      type: "string[]",
+      label: "Follower Ids",
+      description: "A string array of UUIDs of the followers of this story.",
+      optional: true,
+    },
+    groupId: {
+      type: "string",
+      label: "Group Id",
+      description: "The id of the group to associate with this story. A group in Clubhouse API maps to a \"Team\" within the Clubhouse Product.",
+      optional: true,
```

Comment:
That endpoint, the listGroups method, is not wrapped under the Clubhouse SDK for node.js ("clubhouse-lib") that we are using for these action
Here is a screenshot of that failing:
![image](https://user-images.githubusercontent.com/25184117/131576792-fb43c9c2-4fcf-469b-9ef9-9a2c5668dc14.png)
and here a workflow from that came from https://pipedream.com/@sergio/clubhouse-trying-listgroups-with-the-sdk-p_gYC39BO
You can look at the `clubhouse-lib` repo, and there is no `listGroups` listed, there are others such as `listEpics`, `listMembers`, `listTeams` but `listGroups`  is not defined there. Here is the link [to the repo](https://github.com/useshortcut/clubhouse-lib/blob/1dc96890e85b01c7cea2870f3a5397d4fca8ae9c/src/index.ts) if you want to take that look.

This is what I see can be done:
1. Don't use dynamic options.
2. Use the clubhouse API directly, and thus we would have in the codebase mixing API and clubhouse.

Furthermore, the description for group id in the Clubhosue API documentation for "create story" where this `Group ID` prop says

> The id of the group to associate with this story. A group in Clubhouse API maps to a \"Team\" within the Clubhouse Product.
i tried using teams, which exists in the SDK, clubhouse-lib, but creating the story with an id from `listTeams` fails too. here is an screenshot and the [relevant workflow](https://pipedream.com/@sergio/clubhouse-io-create-story-p_8rClzeV):
![image](https://user-images.githubusercontent.com/25184117/131578374-3db2cce0-862c-4873-87d9-f900be16c4f0.png)

So that said, please your input. Should we use the API for this case or don't use `options()` at all, or do you see another way to do this?

Reply:
@sergioeliot2039 you can call listResource() with the URL you want. Or listTeams() may be what you want - I seem to remember that they renamed teams to groups or something.